src/lib.rs

```diff
@@ -91,11 +177,32 @@ fn chd_open(path: String, parent: Option<String>) -> PyResult<Chd> {
     })
 }
 
+#[pyfunction]
+fn try_read_headers(paths: Vec<String>) -> PyResult<Vec<Header>> {
```

Comment:
Rename this to `chd_read_header`

I would rather this take in a single string and return a single header, i.e. 

`fn try_read_header(path: String) -> PyResult<Header>`. Then the for loop can be written in Python or as a list comprehension.

```python
files = [...]
headers = [chd_read_header(f) for f in files]
```

Additionally there's no reason to restrict this to only V5 headers, if the V5 exclusive methods return `Option<&[u8]>` (which will map to `None` in Python for non-V5 headers).